frame.py

```diff
@@ -0,0 +1,211 @@
+
+import cv2 as cv
+import numpy as np
+from registration import Model, register
+from detection import Detector
+
+
+
+
+
+
+
+def detect_features(shot, model, h, w, img1):  #detecting features, matching features
+        MIN_MATCH_COUNT = 10
+        sift = cv.SIFT_create()
+        kp1, des1 = model.kp, model.des
+        #kp1, des1 = sift.detectAndCompute(img1, None)
+        kp2, des2 = sift.detectAndCompute(shot, None)  #detect features
+        FLANN_INDEX_KDTREE = 1
+        index_params = dict(algorithm=FLANN_INDEX_KDTREE, trees=5)
+        search_params = dict(checks=50)
+        flann = cv.FlannBasedMatcher(index_params, search_params)
+
+        matches = flann.knnMatch(des1, des2, k=2)
+
+        good = []
+        for m, n in matches:
+            if m.distance < 0.6 * n.distance:
+                good.append(m)
+
+        if len(good) > MIN_MATCH_COUNT:
+            src_pts = np.float32([kp1[m.queryIdx].pt for m in good]).reshape(-1, 1, 2)
+            dst_pts = np.float32([kp2[m.trainIdx].pt for m in good]).reshape(-1, 1, 2)
+
+            M, mask = cv.findHomography(src_pts, dst_pts, cv.RANSAC, 5.0)
+            matchesMask = mask.ravel().tolist()
+
+            features = []
+
+            ob = []

```

Comment:
Can you please give more meaningful names for variables, did not get the idea of `ob`